lib/sles4sap/console_redirection.pm

```diff
@@ -0,0 +1,174 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::console_redirection;
+use strict;
+use warnings;
+use testapi;
+use Carp qw(croak);
+use Exporter qw(import);
+use serial_terminal qw(select_serial_terminal);
+use Regexp::Common qw(net);
+
+=head1 SYNOPSIS
+
+Library that enables console redirection and file transfers from worker based VM to another host.
+Can be used for cases where worker VM is not the target host for API calls and command execution, but serves only as a jumphost.
+
+=cut
+
+our @EXPORT = qw(
+  connect_target_to_serial
+  disconnect_target_from_serial
+  verify_ip
+  redirection_init
+  check_serial_redirection
+);
+
+my $ssh_opt = '-o StrictHostKeyChecking=no -o ServerAliveInterval=60 -o ServerAliveCountMax=120';
+
+=head2 handle_login_prompt
+
+    handle_login_prompt();
+
+B<ssh_user> Login user
+Detects if login prompt appears and types the password.
+
+=cut
+
+sub handle_login_prompt {
+    my $pwd = get_var('_SECRET_SUT_PASSWORD', $testapi::password);
+    set_serial_term_prompt();
+    my $serial_response = wait_serial(qr/Password:\s*$|:~/i);
+    # Handle password prompt if it appears
+    if (grep /Password:\s*$/, $serial_response) {
+        type_password $pwd;
+        send_key 'ret';
+    }
+    croak 'Failed to setup serial console' unless grep /:~/, $serial_response;
+}
+
+=head2 redirection_init
+
+    redirection_init();
+
+Do preparation before redirecting console. Gets base VM id and initial setup.
+Needs to be done only once.
+
+=cut
+
+sub redirection_init {
+    # This should get worker VM id before any redirection happening
+    # ID serves as identification of the 'base' VM to return to.
+    set_var('WORKER_VM_ID', script_output 'cat /etc/machine-id');
```

Comment:
Who is the consumer of this setting in the future?  Is the only consumer the `croak` in `connect_target_to_serial`?

Reply:
It is used by `check_serial_redirection` as well. 
It's supposed to be run before any ssh jump is done so we know which VM is the local one. 

The whole `redirection_init` will include code for the normal console redirection and file transfer fix as well. 
The normal console will require for example FIFO file to be created and it will go here. This is how it was done on DA project : https://github.com/SUSE/diamond-assurance/blob/e3f3f8be32c5166d24be9fe357a8051e6ca620d1/openqa/da-os-autoinst-tests/lib/da/external_sut_test.pm#L50

Reply:
```
It is used by check_serial_redirection as well.
It's supposed to be run before any ssh jump is done so we know which VM is the local one.

The whole redirection_init will include code for the normal console redirection and file transfer fix as well.
The normal console will require for example FIFO file to be created and it will go here. 
```

Maybe explanation like this can be included in some code comment